src/battle_util.c

```diff
@@ -8304,7 +8304,7 @@ u32 GetMoveTarget(u16 move, u8 setTarget)
     case MOVE_TARGET_FOES_AND_ALLY:
     case MOVE_TARGET_OPPONENTS_FIELD:
         targetBattler = GetBattlerAtPosition(BATTLE_OPPOSITE(GetBattlerSide(gBattlerAttacker)));
-        if (!IsBattlerAlive(targetBattler))
+        if (!IsBattlerAlive(targetBattler) && gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
             targetBattler ^= BIT_FLANK;
```

Comment:
how does this fix the hazard situation? Also wouldn't it be better to remove the battler check for `MOVE_TARGET_OPPONENTS_FIELD` since it targets the side?

Reply:
The function `NoTargetPresent` (present in `attackcanceler`) was causing the issue. Since the target was not alive, it would try to redirect the move to the target's partner, which didn't exist in singles. Can't figure out why having a `gBattlerTarget = 2` would make it target its own side of the field, but it did.

For comparison, here's the current interaction: (the unlabeled parts are the start of `NoTargetPresent` and immediately after the check for the target not being alive)

![pre_fix](https://github.com/user-attachments/assets/ac508ad5-ab9c-4379-adee-36ef424fc3d1)

And here's the fixed interaction: (I also needed to fix another part where it initially set the target to 2)

![with_fix](https://github.com/user-attachments/assets/9e9f84d6-0438-4024-89be-21f1c04dfa70)

I initially left the check in for `MOVE_TARGET_OPPONENTS_FIELD` since I thought it might affect the animation played, but now it doesn't check if the target is alive.

Reply:
thx, I like the second fix more.